zipline/pipeline/mixins.py

```diff
@@ -109,7 +111,14 @@ def _compute(self, windows, dates, assets, mask):
         compute = self.compute
         missing_value = self.missing_value
         params = self.params
-        out = full_like(mask, missing_value, dtype=self.dtype)
+        outputs = self.outputs
+        if outputs is not NotSpecified and len(outputs) > 1:
```

Comment:
If you specify outputs at all, I'd say you should always get a `recarray` with all the fields of your outputs.